### PR TITLE
[Metricbeat] Move mongodb.status metricset to ECS

### PR DESCRIPTION
**Pull request by ruflin**
Only field changed for now is service.version.

The metricset was also changed to use the reporter interface.

### CODE REVIEW

**Comment by ruflin** [comment]
@webmat For now I didn't touch anything around process.

**Comment by ruflin** [comment]
* data.json: I think we adjusted the indentation at ones stage that is why the diff is so large
* timestamp: The timestamp is hardcoded in the generation: https://github.com/elastic/beats/blob/master/metricbeat/mb/testing/data_generator.go#L137

File: metricbeat/module/mongodb/status/_meta/data.json [hunk 1]
```diff
@@ -1,209 +1,207 @@
...
+                "replicated": {
+                    "command": 0,
+                    "delete": 0,
+                    "getmore": 0,
+                    "insert": 0,
+                    "query": 0,
+                    "update": 0
+                }
+            },
+            "process": "mongod",
```
Comment (webmat) [inline metricbeat/module/mongodb/status/_meta/data.json]: To me it sounds superfluous to keep having a custom field here, when we have `process.name` defined in ECS. There's a few places where `process.name` is set, even if there's nothing else being populated in the field set, in the Filebeat modules (system, stack, haproxy).

I don't feel super strongly about it, though.

File: metricbeat/module/mongodb/status/status_integration_test.go [hunk 2]
```diff
@@ -33,33 +32,40 @@ import (
...
+	assert.Empty(t, errs)
+	if !assert.NotEmpty(t, events) {
 		t.FailNow()
 	}
 
-	t.Logf("%s/%s event: %+v", f.Module().Name(), f.Name(), event)
+	t.Logf("%s/%s event: %+v", f.Module().Name(), f.Name(),
+		events[0].BeatEvent("haproxy", "status").Fields.StringToPrint())
+
+	event := events[0].BeatEvent("haproxy", "status").Fields
```
Comment (webmat) [inline metricbeat/module/mongodb/status/status_integration_test.go]: "haproxy" again? Rather "mongodb"? :-)

Comment (ruflin) [reply metricbeat/module/mongodb/status/status_integration_test.go]: That is what happens when working on 2 PR's in parallel. I was so consitent with haproxy that the tests even passed :-)

File: metricbeat/module/mongodb/status/_meta/data.json (hunk 1 above)
Comment (ruflin) [reply metricbeat/module/mongodb/status/_meta/data.json]: Changed it to process.name.

**Comment by ruflin** [comment]
Ready for another round of reviews.

File: metricbeat/module/mongodb/status/status.go [hunk 3]
```diff
@@ -62,20 +57,34 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
...
+	event := mb.Event{}
+	event.MetricSetFields, _ = schema.Apply(result)
+
+	if v, err := event.MetricSetFields.GetValue("version"); err == nil {
+		event.RootFields = common.MapStr{}
+		event.RootFields.Put("service.version", v)
+		event.MetricSetFields.Delete("version")
+	}
+	if v, err := event.MetricSetFields.GetValue("process"); err == nil {
+		event.RootFields = common.MapStr{}
```
Comment (jsoriano) [inline metricbeat/module/mongodb/status/status.go]: Can both `version` and `process` be present? If they are `RootFields` will get overwritten here.

Comment (ruflin) [reply metricbeat/module/mongodb/status/status.go]: yes, good find. Will fix it.

**Pull request merged**